FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_plain_text/DemoTasks/PlaintextMQTTExample.c

```diff
@@ -0,0 +1,656 @@
+		{
+			LogInfo( ( "Publish to the MQTT topic %s.\r\n", mqttexampleTOPIC ) );
+			prvMQTTPublishToTopic( &xMQTTContext );
+
+			/* Process incoming publish echo, since application subscribed to the same
+			 * topic the broker will send publish message back to the application. */
+			LogInfo( ( "Attempt to receive publish message from broker.\r\n" ) );
+			xMQTTStatus = MQTT_ProcessLoop( &xMQTTContext, mqttexamplePROCESS_LOOP_TIMEOUT_MS );
+
+			/* Leave Connection Idle for some time */
+			LogInfo( ( "Keeping Connection Idle.\r\n\r\n" ) );
+			vTaskDelay( mqttexampleDELAY_BETWEEN_PUBLISHES );
+		}
+
+		/************************ Unsubscribe from the topic. **************************/
+		LogInfo( ( "Unsubscribe from the MQTT topic %s.\r\n", mqttexampleTOPIC ) );
+		prvMQTTUnsubscribeFromTopic( &xMQTTContext );
+
+		/* Process Incoming packet from the broker. */
+		xMQTTStatus = MQTT_ProcessLoop( &xMQTTContext, mqttexamplePROCESS_LOOP_TIMEOUT_MS );
+
+		/**************************** Disconnect. ******************************/
+
+		/* Send an MQTT Disconnect packet over the already connected TCP socket.
+		 * There is no corresponding response for the disconnect packet. After sending
+		 * disconnect, client must close the network connection. */
+		LogInfo( ( "Disconnecting the MQTT connection with %s.\r\n", democonfigMQTT_BROKER_ENDPOINT ) );
+		MQTT_Disconnect( &xMQTTContext );
+
+		/* Close the network connection.  */
+		Transport_FreeRTOS_Disconnect( &xNetworkContext );
+
+		/* Wait for some time between two iterations to ensure that we do not
+		 * bombard the public test mosquitto broker. */
+		LogInfo( ( "prvMQTTDemoTask() completed an iteration successfully. Total free heap is %u.\r\n", xPortGetFreeHeapSize() ) );
+		LogInfo( ( "Demo completed successfully.\r\n" ) );
+		LogInfo( ( "Short delay before starting the next iteration.... \r\n\r\n" ) );
+		vTaskDelay( mqttexampleDELAY_BETWEEN_DEMO_ITERATIONS );
+	}
+}
+/*-----------------------------------------------------------*/
+
+static void prvCreateMQTTConnectionWithBroker( MQTTContext_t * pxMQTTContext,
+											   NetworkContext_t * pxNetworkContext )
+{
+MQTTStatus_t xResult;
+MQTTConnectInfo_t xConnectInfo;
+uint16_t usPacketId;
+bool xSessionPresent;
+TransportInterface_t xTransport;
+MQTTApplicationCallbacks_t xCallbacks;
+
+	/***
+	 * For readability, error handling in this function is restricted to the use of
+	 * asserts().
+	 ***/
+
+	/* Fill in Transport Interface send and receive function pointers. */
+	xTransport.pNetworkContext = pxNetworkContext;
+	xTransport.send = Transport_FreeRTOS_send;
+	xTransport.recv = Transport_FreeRTOS_recv;
+
+	/* Application callbacks for receiving incoming published and incoming acks
+	 * from MQTT library. */
+	xCallbacks.appCallback = prvEventCallback;
+	xCallbacks.getTime = prvGetTimeMs;
+
+	/* Initialize MQTT library. */
+	MQTT_Init( pxMQTTContext, &xTransport, &xCallbacks, &xBuffer );
```

Comment:
Better to get the return status and assert.

Reply:
Added assert